lib/Project.js

```diff
@@ -0,0 +1,117 @@
+// @flow
+
+const fs = require('fs');
+const path = require('path');
+const ElmJson = require('./ElmJson');
+
+// Poor man’s type alias. We can’t use /*:: type Project = ... */ because of:
+// https://github.com/prettier/prettier/issues/2597
+const Project /*: {
+  rootDir: string,
+  testsDir: string,
+  generatedCodeDir: string,
+  testsSourceDirs: Array<string>,
+  elmJson: typeof ElmJson.ElmJson,
+} */ = {
+  rootDir: '',
+  testsDir: '',
+  generatedCodeDir: '',
+  testsSourceDirs: [],
+  elmJson: ElmJson.ElmJson,
+};
```

Comment:
Thoughts on making this a class with a constructor?

Reply:
Yes, one – why? 😄

Reply:
I will go and find another hill to die on and leave you in peace.